orc/src/main/java/org/apache/iceberg/orc/ExpressionToSearchArgument.java

```diff
@@ -0,0 +1,355 @@
+
+  static SearchArgument visit(Expression expr, TypeDescription readSchema) {
+    Map<Integer, String> idToColumnName = TypeUtil.visit(ORCSchemaUtil.convert(readSchema), new IdToQuotedColumnName());
+    SearchArgument.Builder builder = SearchArgumentFactory.newBuilder();
+    ExpressionVisitors.visit(expr, new ExpressionToSearchArgument(builder, idToColumnName)).invoke();
+    return builder.build();
+  }
+
+  // Currently every predicate in ORC requires a PredicateLeaf.Type field which is not available for these Iceberg types
+  private static final Set<TypeID> UNSUPPORTED_TYPES = ImmutableSet.of(
+      TypeID.BINARY,
+      TypeID.FIXED,
+      TypeID.UUID,
+      TypeID.STRUCT,
+      TypeID.MAP,
+      TypeID.LIST
+  );
+
+  private SearchArgument.Builder builder;
+  private Map<Integer, String> idToColumnName;
+
+  private ExpressionToSearchArgument(SearchArgument.Builder builder, Map<Integer, String> idToColumnName) {
+    this.builder = builder;
+    this.idToColumnName = idToColumnName;
+  }
+
+  @Override
+  public Action alwaysTrue() {
+    return () -> this.builder.literal(TruthValue.YES);
+  }
+
+  @Override
+  public Action alwaysFalse() {
+    return () -> this.builder.literal(TruthValue.NO);
+  }
+
+  @Override
+  public Action not(Action child) {
+    return () -> {
+      this.builder.startNot();
+      child.invoke();
+      this.builder.end();
+    };
+  }
+
+  @Override
+  public Action and(Action leftChild, Action rightChild) {
+    return () -> {
+      this.builder.startAnd();
+      leftChild.invoke();
+      rightChild.invoke();
+      this.builder.end();
+    };
+  }
+
+  @Override
+  public Action or(Action leftChild, Action rightChild) {
+    return () -> {
+      this.builder.startOr();
+      leftChild.invoke();
+      rightChild.invoke();
+      this.builder.end();
+    };
+  }
+
+  @Override
+  public <T> Action isNull(Bound<T> expr) {
+    return () -> this.builder.isNull(idToColumnName.get(expr.ref().fieldId()),
+        type(expr.ref().type()));
+  }
+
+  @Override
+  public <T> Action notNull(Bound<T> expr) {
+    return () -> this.builder.startNot()
+        .isNull(idToColumnName.get(expr.ref().fieldId()),
+            type(expr.ref().type()))
+        .end();
+  }
+
+  @Override
+  public <T> Action lt(Bound<T> expr, Literal<T> lit) {
+    return () -> this.builder.lessThan(idToColumnName.get(expr.ref().fieldId()),
+        type(expr.ref().type()),
+        literal(expr.ref().type(), lit.value()));
+  }
+
+  @Override
+  public <T> Action ltEq(Bound<T> expr, Literal<T> lit) {
+    return () -> this.builder.lessThanEquals(idToColumnName.get(expr.ref().fieldId()),
+        type(expr.ref().type()),
+        literal(expr.ref().type(), lit.value()));
+  }
+
+  @Override
+  public <T> Action gt(Bound<T> expr, Literal<T> lit) {
+    return () -> this.builder.startNot()
+          .lessThanEquals(idToColumnName.get(expr.ref().fieldId()),
```

Comment:
There is no `greaterThan`? Could you add a comment here to explain that it's using `not("x" <= 5)` instead of `x > 5`?